staticcheck/doc.go

```diff
@@ -443,7 +443,22 @@ falsify results.`,
 	},
 
 	"SA4008": {
-		Title:    `The variable in the loop condition never changes, are you incrementing the wrong variable?`,
+		Title: `The variable in the loop condition never changes, are you incrementing the wrong variable?`,
+		Text: `For example:
+
+	for i := 0; i < 10; j++ { ... }
+
+Additionally, this can may also occur when a loop condition does not have a
+chance to increment, like when the loop body contains an unconditional panic:
+
+	f := func() {
```

Comment:
This example is unfortunate, because Staticcheck will not actually flag it. It doesn't look through the variable. For it to work we need a named function; or you could just call `panic` in the loop.